src/app/test/test-table/test-table.component.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+import { TestListItem } from '../../shared/interfaces/test-list-item';
+import { catchError, Subscription } from 'rxjs';
+import { HttpService } from '../../shared/services/http.service';
+import { ErrorHandling } from '../../shared/classes/error-handling.service';
+import { TabService } from '../../shared/services/tab.service';
+import { HelperService } from '../../shared/services/helper.service';
+import { Report } from '../../shared/interfaces/report';
+import { ReportData } from '../../shared/interfaces/report-data';
+import { View } from '../../shared/interfaces/view';
+import { TestReportsService } from '../test-reports.service';
+import { ToastService } from '../../shared/services/toast.service';
+import { FormsModule } from '@angular/forms';
+
+@Component({
+  selector: 'app-test-table',
+  standalone: true,
+  imports: [FormsModule],
+  templateUrl: './test-table.component.html',
+  styleUrl: './test-table.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TestTableComponent implements OnChanges, OnInit, OnDestroy {
+  @Input() reports: TestListItem[] = new Array<TestListItem>();
+  @Input() currentFilter: string = '';
+  @Input() showStorageIds?: boolean;
+  @Output() toggleCheckEvent: EventEmitter<TestListItem> = new EventEmitter<TestListItem>();
+  @Output() runEvent: EventEmitter<TestListItem> = new EventEmitter<TestListItem>();
```

Comment:
Rerunning a report can also be handled from this component right?

Reply:
The test component calls the run method from the runSelected method, so the method cannot just be moved to the test-table.
I think it will be better if we keep it like this.